plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/proxy/NugetProxyGalleryFacet.java

```diff
@@ -128,12 +128,17 @@ protected void doDestroy() throws Exception {
 
   @Override
   public int count(final String operation, final Map<String, String> parameters) {
+    Iterable<Repository> proxyRepositories = getProxyRepositories();
     final List<Integer> remoteCounts = passQueryToRemoteRepos(nugetQuery(operation, parameters),
-        getProxyRepositories(),
+        proxyRepositories,
         new CountFetcher(fetcher));
-
-    final int hostedCount = count(operation, parameters, getHostedRepositories());
```

Comment:
I wonder if we should change the queries in `StorageTxImpl` to return no results (or count 0) when no repositories are given? Is there ever a situation where passing no repositories into `countAssets`, etc. should query over everything?